### DISCUSSION TITLE
moving from master to main

### DISCUSSION THREAD

**Discussion by Alex-Jordan** [openwebwork/webwork2#1228] [Closed]
We are gearing up to a 2.16 release, which would normally merge `develop` into `master`. What if we took the opportunity to create `main` and do away with `master`? Release notes would include notes about the move. And the wiki documentation would need to be examined for related edits.

https://www.git-tower.com/learn/git/faq/git-rename-master-to-main/

**Comment by pstaabp** [comment]
I whole-hearted agree with this.

**Comment by Alex-Jordan** [comment]
Given that WeBWorK3 is now on people's radar, maybe this is not worth doing with WeBWorK2. How many more releases do we expect? At least 2.17, but more?

**Reply by taniwallach** [reply]
I suspect that webwork2 will be the main system for most users for at least another 2 years... and that even if 2.17 is the last "big" release for webwork2, we will still need to do some ongoing maintenance for the webwork2 code-base for the transition period. Thus having the "modern" name for the main branch is probably a reasonable change.

**Reply by pstaabp** [reply]
Even if webwork2 is going to stay with us for a while, I would push to transition the naming of the main branch though once we get 2.16 out the door.  I'm thinking the only thing we might need to do is some documentation on the wiki using git.

**Reply by taniwallach** [reply]
> ... I would push to transition the naming of the main branch though once we get 2.16 out the door. 

I agree - the release of 2.16 can be to `main` and as you said we just need to fix documentation to refer to `main` instead of master. 

> I'm thinking the only thing we might need to do is some documentation on the wiki using git.

Also some changes in the standard Docker files and the documentation about using Docker on the GitHub Wiki.

**Comment by Alex-Jordan** [comment]
This discussion is in the "Closed" category, but is not actually closed. Closing now.
